xarray/backends/api.py

```diff
@@ -379,6 +380,18 @@ def _chunk_ds(
     return backend_ds._replace(variables)
 
 
+def _create_default_indexes(ds, create_default_indexes):
```

Comment:
Nit: rename to `_maybe_create_default_indexes` will make it slightly clearer that this will not overwrite any preexisting index on a dimension coordinate.